fitbenchmarking/results_output.py

```diff
@@ -133,158 +118,92 @@ def create_results_dict(results_per_test, linked_problems):
         prob_name = name + ' ' + str(count)
         url = link.split('<')[1].split('>')[0]
         html_links.append(template.format(url, prob_name))
-        acc_results[prob_name] = [result.chi_sq for result in prob_results]
-        time_results[prob_name] = [result.runtime for result in prob_results]
-    return acc_results, time_results, html_links
+        results[prob_name] = [result for result in prob_results]
+    return results, html_links
+
+
+def preproccess_data(data):
+    """
+    Helper function that preprocesses data into the right format for printing
+
+    :param data: dictionary of results objects
+    :type data: dict
+    """
+    for results in data.values():
+        min_chi_sq = min([r.chi_sq for r in results])
+        min_runtime = min([r.runtime for r in results])
+        for r in results:
+            r.min_chi_sq = min_chi_sq
+            r.min_runtime = min_runtime
+            r.set_normalised_data()
+            r.set_colour_scale()
 
 
-def create_pandas_dataframe(table_data, minimizers):
+def create_pandas_dataframe(table_data, minimizers, table_suffix):
     """
-    Generates pandas tables.
+    Generates pandas data frame.
 
     :param table_data : dictionary containing results, i.e.,
                             {'prob1': [result1, result2, ...],
                              'prob2': [result1, result2, ...], ...}
     :type table_data : dict
     :param minimizers : list of minimizers (column headers)
-    :type group_name : list
+    :type minimizers : list
+    :param table_suffix : set output to be runtime or accuracy table
+    :type table_suffix : list
 
 
     :return : dict(tbl, tbl_norm, tbl_combined) dictionary of
                pandas DataFrames containing results.
-    :rtype : dict{pandas DataFrame, pandas DataFrame,
-                   pandas DataFrame}
+    :rtype : dict{pandas DataFrame, pandas DataFrame}
     """
+    def select_table(value, table_suffix):
+        '''
+        Selects either accuracy or runtime table
+        '''
+        value.table_type = table_suffix
+        value = copy.copy(value)
```

Comment:
I think the need for copying is unnecessary here. Could this be merged with the render function and put the loop in there?

Reply:
I think its nice to create them first and render after